app/controllers/concerns/error_rendering.rb

```diff
@@ -0,0 +1,37 @@
+module ErrorRendering
+  class ErrorRenderer
+    def initialize(target, source)
+      @target = target
+      @source = source
+    end
+
+    def now(*args, **kwargs)
+      @source.send(:render_err_target, @target.now, *args, **kwargs)
+    end
+  end
+
+  def render_err(chain=:chain, *args, **kwargs)
+    if chain == :chain
+      ErrorRenderer.new(flash, self)
+    else
+      render_err_target(flash, chain, *args, **kwargs)
+    end
+  end
```

Comment:
Quick question, what... is even going on here.

Reply:
clearly me-of-three-years-ago was having too much fun writing the least clear code possible

I'll take another look at this in a bit